brooklyn-server/core/src/main/java/org/apache/brooklyn/entity/group/DynamicFabricImpl.java

```diff
@@ -112,9 +115,10 @@ public void start(Collection<? extends Location> locsO) {
         if (locsO!=null) {
             addLocations(locsO);
         }
-        
+        Collection<Location> locs = Collections.unmodifiableCollection(Locations.getLocationsCheckingAncestors(getLocations(), this));
```

Comment:
move this into the `newLocations.isEmpty()` condition we only need it it `locs0` is empty